### ISSUE TITLE
[Feature] generate Secret/ConfigMap from vela config

### ISSUE THREAD

**Issue by zzxwill** [oam-dev/kubevela#862]
Normally we set container env from ConfigMap or Secret as below.

```
name: hello

services:
  express-server:
    type: webservice
    image: oamdev/postgresql-flask-web-application:v0.1
    env:
      - name: DB_HOST
        valueFrom:
          configMapKeyRef:
            name: rds-config
            key: endpoint
      - name: DB_USER
        valueFrom:
          configMapKeyRef:
            name: rds-config
            key: username
      - name: DB_PASSWORD
        valueFrom:
          configMapKeyRef:
            name: rds-config
            key: password
      ports: 80
```

Sometimes, we might want to use vela config data, as it's very convenient to store key-value data in config by vela core or vela cli. For example, if a workload generate database connection information and store it in config as below.

```
services:
  db:
    type: aliyun-oss
    engine_version: "8.0"
    instance_type: "rds.mysql.c1.large"
    instance_name: "poc"
    account_name: "oam"
    password: "Xyfe83jfewGGf"
    config: "rds-config"
```

It would be useful if there is an option to generate Secret/ConfigMap based on the config data as env can directly retrieve the data form Secret/ConfigMap.

/cc @hongchaodeng

**Comment by hongchaodeng** [comment]
> an option to generate Secret/ConfigMap based on the config data

What's the use case for that?

**Comment by zzxwill** [comment]
> > an option to generate Secret/ConfigMap based on the config data
> 
> What's the use case for that?

For example, https://github.com/zzxwill/catalog/tree/terraform/applications/terraform.

**Comment by hongchaodeng** [comment]
> For example, https://github.com/zzxwill/catalog/tree/terraform/applications/terraform.

Providing a link won't explain anything. Please explain in words with more details of the use case

**Comment by zzxwill** [comment]
@hongchaodeng config could only be used in cue context, ie, in WorkloadDefinition and TraitDefinition, but app developers' focus in Appfile, and normally they will use env to consume data from config. So I wonder whether it's necessary to generate a secret from vela config?

> config 目前是 build 到 cue context 里，这个是在扩展层（definition），但是用 env 的是在抽象层(appfile)。
是不是需要提供一个选项，进一步把 config 生成 secret/cm，这样 env 可以从 secret/cm 获取数据？

**Comment by resouer** [comment]
Please ref to https://docs.docker.com/engine/swarm/secrets/#use-secrets-in-compose about how to define secret/config in appfile and fix this by following PR. We follow exactly same pattern.

**Comment by hongchaodeng** [comment]
> config could only be used in cue context, ie, in WorkloadDefinition and TraitDefinition, but app developers' focus in Appfile, and normally they will use env to consume data from config. So I wonder whether it's necessary to generate a secret from vela config?

What are the user workflow? Are they consuming k8s secrets in their app? Like using volume from secret or env from secret?

**Comment by zzxwill** [comment]
Closing this issue as it's included in #1128
